gateway-ha/src/main/java/io/trino/gateway/ha/resource/GatewayWebAppResource.java

```diff
@@ -423,4 +436,67 @@ public Response readExactMatchSourceSelector()
         List<ResourceGroupsManager.ExactSelectorsDetail> selectorsDetailList = resourceGroupsManager.readExactMatchSourceSelector();
         return Response.ok(Result.ok(selectorsDetailList)).build();
     }
+
+    @GET
+    @RolesAllowed("USER")
+    @Produces(MediaType.APPLICATION_JSON)
+    @Path("/getRoutingRules")
+    public Response readRoutingRules()
+    {
+        String content = null;
+        try {
+            String rulesConfigPath = configuration.getRoutingRules().getRulesConfigPath();
+            content = new String(Files.readAllBytes(Paths.get(rulesConfigPath)));
+            ObjectMapper yamlReader = new ObjectMapper(new YAMLFactory());
+            YAMLParser parser = new YAMLFactory().createParser(content);
+            List<RoutingRules> routingRulesList = new ArrayList<>();
+            while (parser.nextToken() != null) {
+                RoutingRules routingRules = yamlReader.readValue(parser, RoutingRules.class);
+                routingRulesList.add(routingRules);
+            }
+            return Response.ok(Result.ok(routingRulesList)).build();
+        }
+        catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @POST
+    @RolesAllowed("ADMIN")
+    @Consumes(MediaType.APPLICATION_JSON)
```

Comment:
We should document how to use this API for clients other than the web ui

Reply:
Should I add it here - https://github.com/trinodb/trino-gateway/blob/main/docs/gateway-api.md ?

Reply:
yes, that looks perfect